sdk/include/opentelemetry/sdk/resource/resource.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/nostd/unique_ptr.h"
+#include "opentelemetry/sdk/trace/attribute_utils.h"
+#include "opentelemetry/sdk/version/version.h"
+#include "opentelemetry/version.h"
+
+#include <cstdlib>
+#include <memory>
+#include <sstream>
+#include <unordered_map>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace resource
+{
+
+class Resource
+{
+public:
+  Resource(const opentelemetry::common::KeyValueIterable &attributes) noexcept;
+
+  Resource(const opentelemetry::sdk::trace::AttributeMap &attributes) noexcept;
+
+  template <class T,
+            nostd::enable_if_t<common::detail::is_key_value_iterable<T>::value> * = nullptr>
+  Resource(const T &attributes) noexcept : Resource(common::KeyValueIterableView<T>(attributes))
+  {}
+
+  Resource(const std::initializer_list<std::pair<nostd::string_view, common::AttributeValue>>
+               attributes) noexcept
+      : Resource(nostd::span<const std::pair<nostd::string_view, common::AttributeValue>>{
+            attributes.begin(), attributes.end()})
+  {}
```

Comment:
What's your opinion about replacing those with a single:
```c++
Resource(const std::map<std::string, opentelemetry::sdk::trace::SpanDataAttributeValue> attributes) noexcept;
```

As resources are an SDK only concept, we don't need to worry about ABI compatibility here.

Also, we should move the `attribute_utils.h` (containing the `AttributeMap`) from `trace` to `common`. Resources will be used by metrics as well.

Reply:
Good point. I have made it directory using STL class unordered_map now.  And `AttributeMap` is moved to `common` as part of another PR.

Reply:
Good point, 
Using `std::unordered_map<std::string, opentelemetry::sdk::common::OwnedAttributeValue` directly, and getting rid of `AttributeMap`.